BedrockServer.cpp

```diff
@@ -195,21 +195,62 @@ void BedrockServer::sync(SData& args,
         // end up lost in the ether.
         {
             SAUTOLOCK(server._futureCommitCommandMutex);
+
+            // First, see if anything has timed out, and move that back to the main queue.
+            if (server._futureCommitCommandTimeouts.size()) {
+                uint64_t now = STimeNow();
+                auto it =  server._futureCommitCommandTimeouts.begin();
+                while (it != server._futureCommitCommandTimeouts.end() && it->first < now) {
```

Comment:
[They are sorted](https://en.cppreference.com/w/cpp/container/multimap):

> Multimap is an associative container that contains a sorted list of key-value pairs, while permitting multiple entries with the same key. Sorting is done according to the comparison function Compare, applied to the keys. Search, insertion, and removal operations have logarithmic complexity. 

The default Compare function is: `class Compare = std::less<Key>>`.

This allows us only to look at commands we know have timed out, instead of all commands in `_futureCommitCommandTimeouts`.

Reply:
👍